[SHOPWARE-6] Implement Twitter Buttons

nekwo12 opened the pull request:
SHOPWARE-6: Add the ability to integrate twitter buttons.

nekwo12 committed:
[SHOPWARE-6] Add twitter buttons component

Provide essential options in order to switch between the button types.

BrocksiNet commented on src/Resources/public/administration/css/plugin-twitter.css:
```diff
@@ -0,0 +1 @@
+.sw-cms-el-twitter-timeline{height:100%;width:100%;min-height:315px}.sw-cms-el-preview-twitter-timeline{position:relative}.sw-cms-el-preview-twitter-timeline img{margin:0 auto;display:block;max-height:110px}.sw-cms-el-twitter-buttons{height:100%;width:100%;min-height:315px}.sw-cms-el-preview-twitter-buttons{position:relative}.sw-cms-el-preview-twitter-buttons img{margin:0 auto;display:block;max-h ...
```
should we push this file? also the minified js file below?

nekwo12 replied:
no it should not. will be removed in the final version

nekwo12 committed:
[SHOPWARE-6] Add translations and move condition outside of templates

nekwo12 committed:
[SHOPWARE-6] Fix translations

`sw-cms.elements` had to be used

nekwo12 committed:
[SHOPWARE-6] Exclude generated assets

nekwo12 committed:
[SHOPWARE-6] Render template dynanically

Added service to be able to fetch component dependet configuration

BrocksiNet commented on src/Resources/app/administration/src/module/sw-cms/elements/twitter-buttons/config/index.js:
```diff
@@ -0,0 +1,94 @@
+                {
+                    id: 'mention',
+                    name: this.$tc('sw-cms.elements.twitterButtons.mention.label'),
+                    href: this.replaceWithDataAttr('https://twitter.com/intent/tweet?screen_name=__ATTR__&ref_src=twsrc%5Etfw', 'handle'),
```
We should not add **_?ref_src_** here.
It is just for tracking. Or am I wrong?

nekwo12 replied:
I don't really know what you mean. Where do you see `?ref_src`?
The thing that i am seeing is that the href for this button type is missing altogether.

BrocksiNet replied:
After __ATTR__

nekwo12 replied:
The url is taken from the example given by https://publish.twitter.com/#. I will look into it if it can be removed

BrocksiNet commented on src/Resources/app/administration/src/module/sw-cms/snippet/de-DE.json:
```diff
@@ -3,6 +3,52 @@
+          "label": "Twitter Follow Button",
+          "options": {
+            "handle": {
+              "placeholder": "Paste a profile URL or @username"
```
Do not forget to translate ;-)

BrocksiNet commented:
I also tested the backend user flow. I get some errors when I want to add the button in e.g. category layout. When I try to save it get a 404 and save of the page is not possible (content stayes like before). 

**_I see some errors in console like this:_**
`{"errors":[{"code":"2fa2158c-2a7f-484b-98aa-975522539ff8","status":"400","detail":"This field is missing.","template":"This field is missing.","meta":{"parameters":{"{{ field }}":"\u0022source\u0022"}},"source":{"pointer":"\/0\/sections\/0\/blocks\/0\/slots\/0\/translations\/2fbb5fe2e29a4d70aa5854ce7ce3e20b\/config\/twitterButtons\/source"}},{"code":"2fa2158c-2a7f-484b-98aa-975522539ff8","status":"400","detail":"This field is missing.","template":"This field is missing.","meta":{"parameters":{"{{ field }}":"\u0022value\u0022"}},"source":{"pointer":"\/0\/sections\/0\/blocks\/0\/slots\/0\/translations\/2fbb5fe2e29a4d70aa5854ce7ce3e20b\/config\/twitterButtons\/value"}},{"code":"7703c766-b5d5-4cef-ace7-ae0dd82304e9","status":"400","detail":"This field was not expected.","template":"This field was not expected.","meta":{"parameters":{"{{ field }}":"\u0022handle\u0022"}},"source":{"pointer":"\/0\/sections\/0\/blocks\/0\/slots\/0\/translations\/2fbb5fe2e29a4d70aa5854ce7ce3e20b\/config\/twitterButtons\/handle"}},{"code":"7703c766-b5d5-4cef-ace7-ae0dd82304e9","status":"400","detail":"This field was not expected.","template":"This field was not expected.","meta":{"parameters":{"{{ field }}":"\u0022userId\u0022"}},"source":{"pointer":"\/0\/sections\/0\/blocks\/0\/slots\/0\/translations\/2fbb5fe2e29a4d70aa5854ce7ce3e20b\/config\/twitterButtons\/userId"}},{"code":"7703c766-b5d5-4cef-ace7-ae0dd82304e9","status":"400","detail":"This field was not expected.","template":"This field was not expected.","meta":{"parameters":{"{{ field }}":"\u0022buttonType\u0022"}},"source":{"pointer":"\/0\/sections\/0\/blocks\/0\/slots\/0\/translations\/2fbb5fe2e29a4d70aa5854ce7ce3e20b\/config\/twitterButtons\/buttonType"}},{"code":"7703c766-b5d5-4cef-ace7-ae0dd82304e9","status":"400","detail":"This field was not expected.","template":"This field was not expected.","meta":{"parameters":{"{{ field }}":"\u0022buttonOptions\u0022"}},"source":{"pointer":"\/0\/sections\/0\/blocks\/0\/slots\/0\/translations\/2fbb5fe2e29a4d70aa5854ce7ce3e20b\/config\/twitterButtons\/buttonOptions"}}]}`

Is this working on your machine?

nekwo12 commented:
> I also tested the backend user flow. I get some errors when I want to add the button in e.g. category layout. When I try to save it get a 404 and save of the page is not possible (content stayes like before).

Yes it was working for me, but i will look into it.

nekwo12 committed:
[SHOPWARE-6] WIP

- Add node dependencies
- Add button to init twitter script
- Overhaul data structure

TODO:
- Fix save on cms pages

Commit:
Merge branch 'develop' into SHOPWARE-6

Commit:
add twitter lib, fix twitter button save, improve structure

Commit:
sync translation
